subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/metadata/GccMetadataProvider.java

```diff
@@ -144,7 +144,7 @@ private String determineVendor(String error, VersionNumber versionNumber, File g
                     if (isCygwin) {
                         include = mapCygwinPath(cygpathExe, include);
                     }
-                    builder.add(FileUtils.normalize(new File(include)));
+                    builder.add(FileUtils.normalize(new File(include).toPath().toRealPath().toFile()));
```

Comment:
This is a bit tricky. When the path include a symbol link, `FileUtils.normalize` will produce incorrect result, failing the compilation. For example, given the include path:

```
/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9
```

`FileUtils.normalize` outputs `/include/c++/9`, which is incorrect because `/lib` is a symbol link:

```
root@dev149 /usr # realpath /lib/gcc/x86_64-linux-gnu/9/../../../..
/usr
root@dev149 /usr # ls -alth /lib
lrwxrwxrwx 1 root root 7 May 23  2020 /lib -> usr/lib
```

So we have to resolve the symlink to get the correct path.